src/main/java/net/datafaker/service/WeightedRandomSelector.java

```diff
@@ -0,0 +1,122 @@
+package net.datafaker.service;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * A utility class for selecting a random element from a list based on assigned weights.
+ **/
+public class WeightedRandomSelector {
+    private static final String WEIGHT_KEY = "weight";
+    private static final String VALUE_KEY = "value";
+    private double[] cumulativeWeights;
+    private Object[] values;
```

Comment:
How about several (at least 2 threads) working with this
i guess they will concurrently change same class fields and inpact each other

Reply:
@snuyanzin , with my last changes, I made the class stateless. Please, review.

Reply:
i see, however it looks like now again refilling of arrays is happening on each iteration or did I miss anything?